src/main/java/org/truffleruby/core/thread/GetCurrentRubyThreadNode.java

```diff
@@ -24,13 +25,24 @@ public static GetCurrentRubyThreadNode create() {
 
     public abstract DynamicObject executeGetRubyThread(VirtualFrame frame);
 
-    @Specialization(
-            guards = "getCurrentJavaThread(frame) == cachedJavaThread",
-            limit = "getCacheLimit()"
-    )
+    /*
+     * Note: the frame argument is used as a dynamic parameter here. Otherwise, The TruffleDSL
+     * assumes cached parameters or calls with only cached arguments (including calls with no
+     * arguments) never change.
+     */
+    /*
```

Comment:
It's weird to have two comment blocks like this back to back. But I won't make you change it.

Reply:
My IDE merged them initially, so I did that.
Adding an empty line in between seems to workaround that problem, so it's fixed now.